Devil's Punishment v2/Assets/Devil's Punishment Source/Scripts/Player/GunController.cs

```diff
@@ -395,9 +409,17 @@ async Task Fire()
             soundManager.PlaySound("AssaultRifle", "Shot");
         }
         
+		// magic number to adjust also here
 
+		if(isAiming)

```

Comment:
The issue with recoil may be here,
`async Task Fire()` is called multiple time when we fire a gun (we have to see how many times at line `316 GunController.cs`) 
Therefore it keeps adding to the recoil amount while the `Sway()` function lerps it down contiuously as well. But with the current values lerp is slower, but that's fine.
Instead of adding the recoil everrtime we fire, we can have a way to figure out if the current spray is ongoing and if it's ongoing for say 1 or 2 seconds then stop adding to the recoil

Reply:
I think that the recoil should be getting higher the more you shootand recovery at a slower speed if you shoot for longer while not aiming.